### PR TITLE
Fix layout styles not applying in widgets customizer

### PR DESCRIPTION
**Pull request by kevin940726**
## Description

Fix a part of https://github.com/WordPress/gutenberg/issues/35826.

Since https://github.com/WordPress/gutenberg/pull/34493, the Social Links block is now using the experimental `flex` layout system. Turns out it requires the editor to have the `editor-styles-wrapper` class. I don't understand why this is requirement though, IMO this should be handled by one of components we're using implicitly.

However, this only solves a part of the issue. The preview is still not having the styles applied.

image

This seems like a separate issue though. The class name for the `flex` layout is generated, but it's not the same as the styles being injected to the page. For instance, the class name on the DOM is `wp-container-61727099e0b13`, while the CSS has `wp-container-6172708e225e0` instead.

This is probably related to the _realtime preview_ nature of the customizer. Maybe [the code](https://github.com/WordPress/gutenberg/blob/311c96e3c2e4a47ae7dae69fe40100fddfe11430/lib/block-supports/layout.php#L131) that generates the ids somehow ran it 2 separate times resulting into 2 different ids. I'll probably need some help digging into this later, as I know very little about this experiment layout thing.

## How has this been tested?

1. Go to Appearance -> Customize -> Widgets
2. Add a Social Links block and add a random social link
3. The style should be the same as the screenshot below

## Screenshots 

image

## Types of changes

Bug fix

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] I've tested my changes with keyboard and screen readers. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal).

### CODE REVIEW

**Comment by jasmussen** [comment]
Thanks for this PR. For some more context, in the post editor when using social links, the following styles are output for the social links block:

Screenshot 2021-10-22 at 11 07 48

Those styles appear to also be output in the widget editor, but as you note, they don't reach their target because of the missing `editor-styles-wrapper` class. And I can confirm that with that class added, as this PR does, things look right again:
Screenshot 2021-10-22 at 11 14 35

However the fonts are now also all wrong, because suddenly the editor normalization stylesheet is applied, as you can see from the serif font above. Compare with "before":
Screenshot 2021-10-22 at 11 16 31

I'm not sure what the best fix is here, I'd be tempted to ask @youknowriad or @ntsekouras for some input.

However if I reapply a few styles to the widget area, things look fine again. For example if I output this:

```
.customize-control-sidebar_block_editor .editor-styles-wrapper {
    font-family: $default-font;
    font-size: $default-font-size;
}
```

Things look pretty close to "normal" again:

Screenshot 2021-10-22 at 11 21 02

Maybe that's a good quick fix? Add both the class, and some boilerplate CSS?

**Comment by youknowriad** [comment]
The `editor-styles-wrapper` is not added by the layout block support itself. In the editor, the styles are output based on the selector here https://github.com/WordPress/gutenberg/blob/630db5dcabe65dac3cc13808547a520a6d40853c/packages/block-editor/src/hooks/layout.js#L216 and on the backend here https://github.com/WordPress/gutenberg/blob/trunk/lib/block-supports/layout.php#L132 

So there must be something specific here that adds this `.editor-styles-wrapper` to the generated styles. The only thing I can think of is the "block editor styles" behavior (passing `styles` to the block editor settings), maybe in customize-widgets, these styles get passed there for some reason?

**Comment by youknowriad** [comment]
Just did some tests and I do see the `editor-styles-wrapper` generated even in the post editor, I'm not sure where it comes from though 🤔

**Comment by youknowriad** [comment]
Looks like my old self disagrees with me and added a comment to explain why that classname is needed https://github.com/WordPress/gutenberg/commit/bf405a3f1280cade6b9f480c973261b987d5fd53#diff-8c5752e7c71f0bb5c68ff94ad60a13a9d494c189dcb3d501d6364489d66062ddR10-R15

So basically, yeah for now, that classname is mandatory for implementing block editors properly. I guess the fix here is good to go.

**Comment by jasmussen** [comment]
> I guess the fix here is good to go.

I would add that we should add the following CSS somewhere before landing, though:

```
.customize-control-sidebar_block_editor .editor-styles-wrapper {
	font-family: $default-font;
	font-size: $default-font-size;
}
```

I'm not sure what would be a good file to add that to, I would guess whichever component outputs the `.customize-control-sidebar_block_editor` class. But just adding this CSS lets us avoid the times new roman unstyled look. Then perhaps @critterverse can look into whether we need additional tweaks to polish it further.

**Comment by youknowriad** [comment]
@jasmussen if editor styles are not meant to be loaded in this context, we can use the "default editor styles" we use in the post editor sometimes instead of that particular style.

**Comment by jasmussen** [comment]
Yes that seems like an ideal solution.

**Comment by kevin940726** [comment]
Added default styles to the editor!

> styles still not applying in the preview

So do we have any solutions for this yet? Or should we leave it to another PR?

**Comment by jasmussen** [comment]
@kevin940726 Can you elaborate on the preview? In my testing of TT1, the social links show up correctly in the canvas:
Screenshot 2021-10-25 at 08 22 30

File: packages/customize-widgets/src/components/sidebar-block-editor/style.scss
Comment (youknowriad) [inline packages/customize-widgets/src/components/sidebar-block-editor/style.scss]: Why can't we load that file directly?

Comment (youknowriad) [reply packages/customize-widgets/src/components/sidebar-block-editor/style.scss]: I mean as a "styles" settings in the block editor provider.

**Comment by kevin940726** [comment]
@jasmussen It's not when you edit it.

https://user-images.githubusercontent.com/7753001/138995501-29cf7288-6d40-4e68-96ee-b89dca50e3cf.mp4

File: packages/customize-widgets/src/components/sidebar-block-editor/style.scss
Comment (kevin940726) [reply packages/customize-widgets/src/components/sidebar-block-editor/style.scss]: Because... I didn't know it's a thing? 😅

**Review by youknowriad — APPROVED** [review]
The changes here make sense to me, though I didn't test very well.

**Pull request merged**
